src/plugins/amqplib.js

```diff
@@ -28,12 +28,16 @@ function createWrapDispatchMessage (tracer, config) {
       addTags(this, tracer, config, span, 'basic.deliver', fields)
 
       setImmediate(() => {
-        tracer.scopeManager().activate(span, true)
+        tracer.scopeManager().activate(span, process.env.DD_CONTEXT_PROPAGATION !== 'false')
```

Comment:
I don't really understand this change. Why would closing the span on finish be dependent on the context propagation being enabled or not? Maybe I'm missing something.